core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -437,6 +445,61 @@ private Block buildBlock(Type type, List<BlockView> inputs, BlockBuilderStatus b
         return blockBuilder.build();
     }
 
+    enum TestType
+    {
+        BIGINT(BigintType.BIGINT),
+        BOOLEAN(BooleanType.BOOLEAN),
+        INTEGER(IntegerType.INTEGER),
+        CHAR_10(createCharType(10)),
+        VARCHAR(createUnboundedVarcharType()),
+        DOUBLE(DoubleType.DOUBLE),
+        SMALLINT(SmallintType.SMALLINT),
+        TINYINT(TinyintType.TINYINT),
+        VARBINARY(VarbinaryType.VARBINARY),
+        LONG_DECIMAL(createDecimalType(Decimals.MAX_SHORT_PRECISION + 1)),
+        LONG_TIMESTAMP(createTimestampType(9)),
+        ROW_BIGINT_VARCHAR(anonymousRow(BigintType.BIGINT, VarcharType.VARCHAR)),
+        ARRAY_BIGINT(new ArrayType(BigintType.BIGINT), false),
+        VARCHAR_WITH_TEST_BLOCK(VarcharType.VARCHAR, TestVariableWidthBlock.adaptation(), true);
+
+        private final Type type;
+        private final Function<Block, Block> blockAdaptation;
+
+        private final boolean supportsNullRle;
+
+        TestType(Type type)
```

Comment:
public (and below)

Reply:
enum constructors have to be private

Reply:
Make it private then (not package private)

Reply:
it is private. one does not simply change the enum constructor visibility :-)